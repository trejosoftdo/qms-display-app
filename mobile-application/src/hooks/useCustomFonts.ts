import { useEffect, useState } from "react";
import { useFonts } from 'expo-font';
import * as SplashScreen from 'expo-splash-screen';
import { DEFAULT_LIGHT_FONT_NAME, DEFAULT_MEDIUM_FONT_NAME, DEFAULT_REGULAR_FONT_NAME } from "../common/constants";


export type FontsResult = {
  fontsLoaded: boolean;
};

SplashScreen.preventAutoHideAsync();


/**
 * Hook used to load custom fonts
 * 
 * @returns FontsResult
 */
const useCustomFonts = (): FontsResult => {
  const [hidden, setHidden] = useState(false);
  const [fontsLoaded] = useFonts({
    [DEFAULT_REGULAR_FONT_NAME]: require('../../assets/fonts/Montserrat-Regular.ttf'),
    [DEFAULT_MEDIUM_FONT_NAME]: require('../../assets/fonts/Montserrat-Medium.ttf'),
    [DEFAULT_LIGHT_FONT_NAME]: require('../../assets/fonts/Montserrat-Light.ttf'),
  });

  useEffect(() => {
    if (fontsLoaded) {
      SplashScreen.hideAsync()
        .then(() => setHidden(true));
    }
  }, [fontsLoaded]);

  
  return {
    fontsLoaded: fontsLoaded && hidden,
  };
};

export default useCustomFonts;
