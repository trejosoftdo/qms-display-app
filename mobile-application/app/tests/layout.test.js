import React from 'react';
import useCustomFonts from '../../src/hooks/useCustomFonts';

jest.mock('../../i18n');
jest.mock('../../src/hooks/useCustomFonts');

import Layout from '../_layout';

describe('Layout component', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders the component as expected when the fonts are loaded', () => {
    useCustomFonts.mockReturnValue({ fontsLoaded: true });
    expect(renderShallow(<Layout />)).toMatchSnapshot();
  });

  it('does not render the component when the fonts are not loaded', () => {
    useCustomFonts.mockReturnValue({ fontsLoaded: false });
    expect(renderShallow(<Layout />)).toMatchSnapshot();
  });
});
