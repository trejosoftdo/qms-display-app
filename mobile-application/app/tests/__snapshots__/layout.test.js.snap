// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component does not render the component when the fonts are not loaded 1`] = `null`;

exports[`Layout component renders the component as expected when the fonts are loaded 1`] = `
<PaperProvider
  theme={
    {
      "animation": {
        "scale": 1,
      },
      "colors": {
        "backdrop": "rgba(50, 47, 55, 0.4)",
        "background": "#ffffff",
        "elevation": {
          "level0": "transparent",
          "level1": "rgb(247, 243, 249)",
          "level2": "rgb(243, 237, 246)",
          "level3": "rgb(238, 232, 244)",
          "level4": "rgb(236, 230, 243)",
          "level5": "rgb(233, 227, 241)",
        },
        "error": "#cc426a",
        "errorContainer": "rgba(249, 222, 220, 1)",
        "inverseOnSurface": "rgba(244, 239, 244, 1)",
        "inversePrimary": "rgba(208, 188, 255, 1)",
        "inverseSurface": "rgba(49, 48, 51, 1)",
        "onBackground": "rgba(28, 27, 31, 1)",
        "onError": "rgba(255, 255, 255, 1)",
        "onErrorContainer": "rgba(65, 14, 11, 1)",
        "onPrimary": "rgba(255, 255, 255, 1)",
        "onPrimaryContainer": "rgba(33, 0, 93, 1)",
        "onSecondary": "rgba(255, 255, 255, 1)",
        "onSecondaryContainer": "rgba(29, 25, 43, 1)",
        "onSurface": "rgba(28, 27, 31, 1)",
        "onSurfaceDisabled": "rgba(28, 27, 31, 0.38)",
        "onSurfaceVariant": "rgba(73, 69, 79, 1)",
        "onTertiary": "rgba(255, 255, 255, 1)",
        "onTertiaryContainer": "rgba(49, 17, 29, 1)",
        "outline": "#d4d4d4",
        "outlineVariant": "rgba(202, 196, 208, 1)",
        "primary": "#2929cc",
        "primaryContainer": "#b4c5e4",
        "scrim": "rgba(0, 0, 0, 1)",
        "secondary": "#3c3744",
        "secondaryContainer": "rgba(232, 222, 248, 1)",
        "shadow": "rgba(0, 0, 0, 1)",
        "surface": "rgba(255, 251, 254, 1)",
        "surfaceDisabled": "rgba(28, 27, 31, 0.12)",
        "surfaceVariant": "rgba(231, 224, 236, 1)",
        "tertiary": "rgba(125, 82, 96, 1)",
        "tertiaryContainer": "#feeef4",
      },
      "dark": false,
      "fonts": {
        "bodyLarge": {
          "fontFamily": "Montserrat-Regular",
          "fontSize": 16,
          "fontWeight": "400",
          "letterSpacing": 0.15,
          "lineHeight": 24,
        },
        "bodyMedium": {
          "fontFamily": "Montserrat-Regular",
          "fontSize": 14,
          "fontWeight": "400",
          "letterSpacing": 0.25,
          "lineHeight": 20,
        },
        "bodySmall": {
          "fontFamily": "Montserrat-Regular",
          "fontSize": 12,
          "fontWeight": "400",
          "letterSpacing": 0.4,
          "lineHeight": 16,
        },
        "default": {
          "fontFamily": "Montserrat-Regular",
          "fontWeight": "normal",
        },
        "headlineLarge": {
          "fontFamily": "Montserrat-Regular",
          "fontSize": 32,
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 40,
        },
        "headlineMedium": {
          "fontFamily": "Montserrat-Regular",
          "fontSize": 28,
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 36,
        },
        "headlineSmall": {
          "fontFamily": "Montserrat-Regular",
          "fontSize": 24,
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 32,
        },
        "labelLarge": {
          "fontFamily": "Montserrat-Regular",
          "fontSize": 14,
          "fontWeight": "500",
          "letterSpacing": 0.1,
          "lineHeight": 20,
        },
        "labelMedium": {
          "fontFamily": "Montserrat-Regular",
          "fontSize": 12,
          "fontWeight": "500",
          "letterSpacing": 0.5,
          "lineHeight": 16,
        },
        "labelSmall": {
          "fontFamily": "Montserrat-Regular",
          "fontSize": 11,
          "fontWeight": "500",
          "letterSpacing": 0.5,
          "lineHeight": 16,
        },
        "light": {
          "fontFamily": "Montserrat-Light",
          "fontWeight": "normal",
        },
        "medium": {
          "fontFamily": "Montserrat-Medium",
          "fontWeight": "normal",
        },
        "regular": {
          "fontFamily": "Montserrat-Regular",
          "fontWeight": "normal",
        },
        "thin": {
          "fontFamily": "Montserrat-Light",
          "fontWeight": "normal",
        },
        "titleLarge": {
          "fontFamily": "Montserrat-Regular",
          "fontSize": 22,
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 28,
        },
        "titleMedium": {
          "fontFamily": "Montserrat-Regular",
          "fontSize": 16,
          "fontWeight": "500",
          "letterSpacing": 0.15,
          "lineHeight": 24,
        },
        "titleSmall": {
          "fontFamily": "Montserrat-Regular",
          "fontSize": 14,
          "fontWeight": "500",
          "letterSpacing": 0.1,
          "lineHeight": 20,
        },
      },
      "isV3": true,
      "roundness": 4,
      "version": 3,
    }
  }
>
  <ForwardRef />
</PaperProvider>
`;
